app/v1/asset/keeper.go

```diff
@@ -76,8 +78,39 @@ func (k Keeper) CreateGateway(ctx sdk.Context, msg MsgCreateGateway) (sdk.Tags,
 
 // EditGateway edits the specified gateway
 func (k Keeper) EditGateway(ctx sdk.Context, msg MsgEditGateway) (sdk.Tags, sdk.Error) {
+	// get the destination gateway
+	gateway, err := k.GetGatewayByMoniker(ctx, msg.Moniker)
+	if err != nil {
+		return nil, err
+	}
+
+	// check if the given owner matches with the owner of the destination gateway
+	if !msg.Owner.Equals(gateway.Owner) {
+		return nil, ErrInvalidOwner(k.codespace, fmt.Sprintf("the address %d is not the owner of the gateway %s", msg.Owner, msg.Moniker))
+	}
+
+	// update the gateway
+	gateway.Identity = msg.Identity
+	gateway.Details = msg.Details
+	gateway.Website = msg.Website
+	gateway.RedeemAddr = msg.RedeemAddr
+	gateway.Operators = msg.Operators
```

Comment:
It is difficult to use if you only want to change one parameter.

Reply:
> It is difficult to use if you only want to change one parameter.

it's up to clients to fill the existing values.